### PR TITLE
Webassembly

### PR DESCRIPTION
**Pull request by anton98i**
Support for compile with Emscripten with `make webassembly`

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -32,6 +32,7 @@ OPTIONS
   --thread -g ${CMAKE_VALA_OPTS} ${VAPIDIRS_ARGS}
 )
 
+if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL Emscripten)
 vala_precompile(TRV_VALA_C
```
Comment (gerhardgruber) [inline src/CMakeLists.txt]: Einrückung

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -40,16 +41,19 @@ PACKAGES
...
 add_library(
   ${DMTESTLIB_NAME} SHARED
   ${DMTESTLIB_VALA_C}
 )
 
+if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL Emscripten)
 add_executable(
   trv
   ${TRV_VALA_C}
 )
```
Comment (gerhardgruber) [inline src/CMakeLists.txt]:
```suggestion
  add_executable(
    trv
    ${TRV_VALA_C}
  )
```

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -71,12 +75,14 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/libdmtestlib.pc.in
   ${CMAKE_CURRENT_BINARY_DIR}/${DMTESTLIB_NAME}.pc @ONLY)
 
 # Install test result viewer
+if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL Emscripten)
 install(
 TARGETS
   trv
 RUNTIME DESTINATION
   bin
 )
```
Comment (gerhardgruber) [inline src/CMakeLists.txt]:
```suggestion
  install(
  TARGETS
    trv
  RUNTIME DESTINATION
    bin
  )
```

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -85,6 +91,16 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL Windows)
     DESTINATION
       bin/
     )
+elseif(${CMAKE_SYSTEM_NAME} STREQUAL Emscripten)
+# Install shared object
```
Comment (gerhardgruber) [inline src/CMakeLists.txt]:
```suggestion
  # Install shared object
```

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -85,6 +91,16 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL Windows)
...
+  install(
+    TARGETS
+      ${DMTESTLIB_NAME}
+    LIBRARY DESTINATION
+    lib/
```
Comment (gerhardgruber) [inline src/CMakeLists.txt]:
```suggestion
      lib/
```

File: src/CMakeLists.txt [hunk 6]
```diff
@@ -85,6 +91,16 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL Windows)
...
+    ARCHIVE DESTINATION
+    lib/
```
Comment (gerhardgruber) [inline src/CMakeLists.txt]:
```suggestion
      lib/
```

**Review by gerhardgruber — CHANGES_REQUESTED** [review]

File: src/CMakeLists.txt (hunk 1 above)
Comment (anton98i) [reply src/CMakeLists.txt]: ok

**Pull request merged**